apps/glusterfs/multi_host_error.go

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright (c) 2019 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"fmt"
+	"strings"
+)
+
+type MultiHostError struct {
+	prefix string
+	errors map[string]error
+}
+
+// NewMultiHostError returns a MultiHostError with the given
+// prefix text. Prefix text will be used in the error string if
+// more than one error is captured.
+func NewMultiHostError(p string) *MultiHostError {
```

Comment:
You could instead require an initial error or a non-empty error list as an argument to avoid an error with an empty list.